doc/_scripts/software_maturity_scanner.py

```diff
@@ -0,0 +1,554 @@
+        cc: Azure container client.
+    """
+
+    data = yaml.safe_dump(data)
+    bc = cc.get_blob_client(filename)
+
+    try:
+        logger.info("Uploading software maturity level information...")
+        bc.upload_blob(data, overwrite=True)
+        logger.info("Upload complete")
+    except ClientAuthenticationError:
+        logger.error("Could not upload to Azure blob storage, invalid access keys")
+
+
+def find_experimental_symbols(logs: List[str]) -> Dict[str, Set[str]]:
+    """Search through a list of files for experimental symbols.
+
+    Return:
+        A mapping from each SoC to a set of symbols that trigger the
+        experimental warning when built with that SoC.
+    """
+
+    socs = {}
+    for log in logs:
+        path = Path(log)
+        content = path.read_text()
+        matches = re.findall(r"Experimental symbol ([A-Z0-9_]+) is enabled", content)
+        if matches:
+            config_path = path.parent / "zephyr" / ".config"
+            if not config_path.exists():
+                logger.warning(f"path '{config_path.as_posix()}' does not exist")
+                continue
+            kconf = kconfiglib.Kconfig(filename=config_path)
+            soc = kconf.variables["CONFIG_SOC"].value
+            soc = soc.strip('"').split("_")[0].lower()
+            for symbol in matches:
+                symbol = "CONFIG_" + symbol
+                if soc not in socs:
+                    socs[soc] = set()
+                socs[soc].add(symbol)
+    return socs
+
+
+def find_all_socs(all_samples: List[str]) -> List[str]:
+    """Search sample.yaml files to find all SoCs currently in use.
+
+    The ``integration_platforms`` field under common and each test is used to
+    find all possible boards, and the SoCs are derived from the board names.
+
+    Any board name that is not nRF, or any board of the nRF51 series is
+    excluded from the returned set.
+
+    Return:
+        A sorted list of all unique SoC names
+    """
+
+    all_boards = set()
+    for sample_path in all_samples:
+        try:
+            with open(sample_path) as sample:
+                content = yaml.safe_load(sample)
+        except (yaml.YAMLError, FileNotFoundError) as err:
+            logger.error(err)
+            continue
+
+        # Add common integration platforms
+        common_platforms = []
+        if "common" in content and "integration_platforms" in content["common"]:
+            common_platforms = content["common"]["integration_platforms"]
+
+        # Add each test's integration platforms
+        for test in content["tests"].values():
+            integration_platforms = common_platforms
+            if "integration_platforms" in test:
+                integration_platforms = test["integration_platforms"]
+            all_boards.update(integration_platforms)
+
+    # Only use nRF boards, but not nRF51 series
+    all_boards = set(
+        filter(lambda b: b.startswith("nrf") and not b.startswith("nrf51"), all_boards)
```

Comment:
do we expect that filter might change in time? If so, it will be good to have them stored in global variable

Reply:
A `BOARD_FILTER` constant has been added as a global variable, and now uses regex to find "_nrfxxx" instead of relying on string start.